clients/src/main/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManager.java

```diff
@@ -538,16 +539,16 @@ public ConsumerGroupHeartbeatRequestData buildRequestData() {
                 }
             });
 
-            // RebalanceTimeoutMs - only sent if has changed since the last heartbeat
-            if (sentFields.rebalanceTimeoutMs != rebalanceTimeoutMs) {
+            // RebalanceTimeoutMs - only sent when joining or if has changed since the last heartbeat
+            if (membershipManager.memberEpoch() == 0 || sentFields.rebalanceTimeoutMs != rebalanceTimeoutMs) {
```

Comment:
If not mistaken, the `rebalanceTimeoutMs` timeout is a static config so we could actually set it only in when joining the group (when epoch == 0).

Reply:
it is static, taken from the `max.poll.interval.ms`, so agree we could set it only when epoch == 0

Reply:
true, this can be simplified